### PR TITLE
⭐️  Adding a new function to fetch the delivery channel and related info (aws)

### PR DESCRIPTION
**Pull request by HRouhani**
Screenshot from 2024-02-26 15-41-49

### CODE REVIEW

File: providers/aws/resources/aws.lr.manifest.yaml [hunk 1]
```diff
@@ -679,6 +679,10 @@ resources:
       desc: |
         Use the `aws.config` resource to assess the configuration of the AWS Config service. The resource provides the `.recorders` field, which returns a list of `aws.config.recorder` resources representing all AWS Config recorders configured across all enabled regions, as well as the `.rules` field, which returns a list of `aws.config.rule` resources representing all AWS Config rules configured across all enabled regions in the account.
     fields:
+      deliveryChannels:
+        min_mondoo_version: latest
```
Comment (tas50) [inline providers/aws/resources/aws.lr.manifest.yaml]:
```suggestion
        min_mondoo_version: 9.0.0
```

File: providers/aws/resources/aws.lr.manifest.yaml [hunk 2]
```diff
@@ -679,6 +679,10 @@ resources:
...
+      deliverychannels:
+        min_mondoo_version: latest
```
Comment (tas50) [inline providers/aws/resources/aws.lr.manifest.yaml]:
```suggestion
        min_mondoo_version: 9.0.0
```

File: providers/aws/resources/aws.lr.manifest.yaml [hunk 3]
```diff
@@ -704,6 +708,22 @@ resources:
           recording == true && lastStatus == "SUCCESS"
         )
       title: Ensure AWS Config is enabled in all regions
+  aws.config.deliverychannel:
+    docs:
+      desc: "The `aws.config.deliverychannel` resource provides fields representing
+        an individual AWS Config delivery channel configured within an account. For
+        usage, read the `aws.config` resource documentation. \n"
```
Comment (tas50) [inline providers/aws/resources/aws.lr.manifest.yaml]:
```suggestion
        usage, read the `aws.config` resource documentation.\n"
```

File: providers/aws/resources/aws.lr [hunk 4]
```diff
@@ -2553,6 +2555,27 @@ private aws.config.recorder @defaults("name region") {
...
+  s3BucketName string
+  // Prefix for the S3 bucket where configuration snapshots are delivered
+  s3KeyPrefix string
+  // ARN of the SNS topic that AWS Config delivers notifications to
+  snsTopicARN string
+  // Region for the delivery channel
+  region string
+}
+
+
```
Comment (tas50) [inline providers/aws/resources/aws.lr]: @HRouhani Can you delete the empty lines here?

Comment (HRouhani) [reply providers/aws/resources/aws.lr]: sure, no worries

**Review by vjeffrey — APPROVED** [review]

**Pull request merged**